graph/memstore/quadstore.go

```diff
@@ -117,17 +116,17 @@ func (qs *QuadStore) ApplyDeltas(deltas []graph.Delta, ignoreOpts graph.IgnoreOp
 		case graph.Add:
 			if !ignoreOpts.IgnoreDup {
 				if _, exists := qs.indexOf(d.Quad); exists {
-					return graph.ErrQuadExists
+					return &graph.QuadError{Action: d.Action, Quad: d.Quad, Err: graph.ErrQuadExists}
```

Comment:
I think this is a bad idea - caller may already compare error with ErrQuadExists. Why to wrap it?

I thinks it's better to remove definition of ErrQuadExists as a variable and make it an error type with single Quad field. This will break code, but at least will show to the user that some changes are necessary.